node/full.go

```diff
@@ -228,12 +232,31 @@ func initDALC(nodeConfig config.NodeConfig, dalcKV ds.TxnDatastore, logger log.L
 	if err != nil {
 		return nil, fmt.Errorf("error decoding namespace: %w", err)
 	}
-	daClient := goDAProxy.NewClient()
-	err = daClient.Start(nodeConfig.DAAddress, grpc.WithTransportCredentials(insecure.NewCredentials()))
+
+	u, err := url.Parse(nodeConfig.DAAddress)
 	if err != nil {
-		return nil, fmt.Errorf("error while establishing GRPC connection to DA layer: %w", err)
+		return nil, fmt.Errorf("error parsing DA address url: %w", err)
+	}
+
+	var daImpl goDA.DA
+	switch u.Scheme {
+	case "grpc":
+		daClient := grpcDA.NewClient()
+		if err := daClient.Start(u.Host, grpc.WithTransportCredentials(insecure.NewCredentials())); err != nil {
+			return nil, fmt.Errorf("error while establishing connection to DA layer: %w", err)
+		}
+		daImpl = daClient
+	case "http", "https":
+		daClient, err := jsonrpcDA.NewClient(context.Background(), nodeConfig.DAAddress, nodeConfig.DAAuthToken)
+		if err != nil {
+			return nil, fmt.Errorf("error while establishing connection to DA layer: %w", err)
+		}
+		daImpl = &daClient.DA
+	default:
+		return nil, fmt.Errorf("error while establishing connection to DA layer: unknown url scheme '%s'", u.Scheme)
 	}
```

Comment:
IMHO this should be extracted as method and moved to DA package.

Reply:
Done.

Reply:
https://github.com/rollkit/go-da/pull/54/commits/f70db8efa5a2719704978d38fe0339abbdad9b6e